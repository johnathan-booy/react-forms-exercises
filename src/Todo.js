import React from "react";

const Todo = ({ id, task, deleteTodo }) => {
	const handleDelete = (e) => {
		deleteTodo({ id, task });
	};
	return (
		<li className="Todo">
			<span>{task}</span>
			<button className="Todo-delete" onClick={handleDelete}>
				X
			</button>
		</li>
	);
};

export default Todo;
