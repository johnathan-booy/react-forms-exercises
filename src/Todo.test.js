import { render, asFragment } from "@testing-library/react";
import Todo from "./Todo";

const TODO = <Todo id={1} task="Get the mail" />;
it("renders without crashing", () => {
	render(<Todo />);
});

it("should match snapshot", () => {
	const { asFragment } = render(TODO);
	expect(asFragment()).toMatchSnapshot();
});

it("should display task", () => {
	const { getByText } = render(TODO);
	expect(getByText("Get the mail")).toBeInTheDocument();
});

it("should display delete button", () => {
	const { getByText } = render(TODO);
	expect(getByText("X")).toBeInTheDocument();
});
