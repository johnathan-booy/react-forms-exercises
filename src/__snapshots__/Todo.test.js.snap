// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <li
    class="Todo"
  >
    <span>
      Get the mail
    </span>
    <button
      class="Todo-delete"
    >
      X
    </button>
  </li>
</DocumentFragment>
`;
